travis/check_tags.py

```diff
@@ -0,0 +1,52 @@
+import re
+import requests
+
+
+def get_commits(travis_repo_slug, travis_pull_request_number, travis_branch, version):
+    development_tags = [':memo:', ':fire:', ':fire_engine:', ':tv:', ':lock:', ':bath:', ':green_heart:', ':cat:', ':bomb:']
+    release_tags = [':tada:', ':zap:', ':sparkles:', ':rainbow:', ':ambulance:', ':heart_eyes:', ':cherries:', ':book:', ':euro:', ':handshake:', ':shield:', ':arrow_up:', ':arrow_down:', ':x:', ':sos:', ':peace_symbol:']
+    verion_tags = {'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}
+    requirements_tags_of_version = [':x:', ':arrow_up:', ':arrow_down:', ':sos:', ':tada:']
+
+    is_addons_dev = re.search(r'addons-dev', str(travis_repo_slug))
+    symbol_in_branch = re.search(r'-', str(travis_branch))
+
+    #GET /repos/:owner/:repo/pulls/:pull_number/commits
+    #See API Github: https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
+    real_errors = {}
+    if travis_pull_request_number:
+        # GET / repos /: owner /:repo / commits
+        commits = requests.get('https://api.github.com/repos/%s/pulls/%s/commits' % (str(travis_repo_slug), str(travis_pull_request_number))).json()
+        for commit in commits:
+            commit = commit.get('commit').get('message')
+            print('Commit: %s' % commit)
+            if is_addons_dev or symbol_in_branch:
+                if any(tag in commit for tag in development_tags):
+                    continue
+                else:
+                    error = {'Wrong commit: ' + commit: 'For addons-dev must be Development tags!'}
+                    real_errors.update(error)
+                    continue
+            if not symbol_in_branch:
+                if any(tag in commit for tag in requirements_tags_of_version):
+                    if any(tag in commit for tag in verion_tags.keys()):
+                        list_digits = [x for x in commit.split(':') if x in verion_tags.keys()]
+                        version_in_commit = ''
+                        for digit in list_digits:
+                            if digit in verion_tags.keys():
+                                version_in_commit += verion_tags.get(digit)
+                        if version.replace(".", "") != version_in_commit:
+                            error = {'Wrong commit: ' + commit: 'Version in commit is wrong!'}
+                            real_errors.update(error)
```

Comment:
The script doesn't check proper order of the tags:


> Version tags has to be used after key tags, for example :arrow_up::one::two: .

Reply:
+